`Fix: Implement Naira symbol display (Closes #24)`

ogeobubu opened the pull request:
This PR addresses [Issue #24 - "Add the currency symbol anywhere an amount or price is displayed"](https://github.com/your-repo/your-project/issues/24) by implementing comprehensive currency formatting improvements:

1. **Currency Display Fixes**:
   - Implemented proper Naira (₦) symbol display throughout the invoice as requested
   - Used the Currency component for consistent formatting (₦3,000 format)
   - Ensured symbol appears in all amount displays (table view and totals section)

2. **Price Calculation**:
   - Fixed getItemPrice to properly access nested price structures
   - Corrected price type access logic (adultFemalePrice/adultMalePrice/childrenPrice)

3. **Data Handling**:
   - Added validation to handle potential data issues
   - Added fallback values (₦0) for missing prices

**Testing:**  
Verified that:
✅ All prices display with ₦ symbol (₦3,000 format) 
✅ Calculations remain accurate with symbol implementation  
✅ Different price types display correctly  
✅ Zero values show as ₦0  

**Before/After Examples** (per issue requirements):
```diff
- Before: 3000
+ After: ₦3,000

- Before: 2500 NGN
+ After: ₦2,500
```

This implementation fully addresses the requirements specified in #24 @natemartins, providing consistent currency symbol display across all amount fields in the invoice system.

**Additional Notes:**
- This PR focused on invoice list page, invoice detail page to render data appropriately.

natemartins reviewed (changes requested):
Please, work on this issue - https://finessecodes.slack.com/archives/D03232HAP97/p1744230879729229 - in this PR

ogeobubu commented:
I've refactored the pricing logic to properly handle all three laundry service types (`ordinary`, `washAndIron`, `ironOnly`) while preventing NaN values. Key changes include:

1. **Price Handling**
   - Created a `getPrice` helper function that safely accesses price values with a fallback to 0
   - Added validation to ensure all price values are numbers
   - Implemented defensive checks for missing/undefined price fields

2. **Service Type Support**
   - Restructured the logic to explicitly handle all three service types
   - Updated the laundry options modal to clearly present these choices
   
![Screenshot 2025-04-09 22 59 50](https://github.com/user-attachments/assets/a4978be6-ea6b-4156-8e66-9d58735a53a5)

   - Maintained backward compatibility with existing data structures

3. **Error Prevention**
   - Added NaN checks with automatic fallback to 0
   - Added fallback IDs when price objects are incomplete

The changes maintain all existing functionality while making the system more robust against invalid price data. All pricing calculations now have safe fallbacks while still allowing the user to identify and fix any underlying data issues through the enhanced logging.

Would appreciate your review of:
- The price fallback strategy
- The service type handling architecture
- Any additional validation we should add

@natemartins
